### PR TITLE
Option to fail the build when there are breaks

### PR DESCRIPTION
**Pull request by jcagme**
Fixes: https://github.com/dotnet/arcade/issues/5528

### CODE REVIEW

File: eng/common/sdl/execute-all-sdl-tools.ps1 [hunk 1]
```diff
@@ -106,6 +107,11 @@ try {
       ExitWithExitCode 1
     }
   }
+
+  if ($BreakOnFailure) {
```
Comment (adiaaida) [inline eng/common/sdl/execute-all-sdl-tools.ps1:111]: Will this fail whether or not guardian actually failed above?

Comment (jcagme) [reply eng/common/sdl/execute-all-sdl-tools.ps1:111]: If there are exceptions then `ExitWithExitCode 1` will be thrown. This only breaks the build if there were Guardian related errors

Comment (adiaaida) [reply eng/common/sdl/execute-all-sdl-tools.ps1:111]: And if there were no guardian related errors?

Comment (jcagme) [reply eng/common/sdl/execute-all-sdl-tools.ps1:111]: It won't break

Comment (sunandabalu) [inline eng/common/sdl/execute-all-sdl-tools.ps1:111]: Are you setting this to true somewhere?

Comment (adiaaida) [reply eng/common/sdl/execute-all-sdl-tools.ps1:111]: Just checking

Comment (jcagme) [reply eng/common/sdl/execute-all-sdl-tools.ps1:111]: Yes https://dnceng.visualstudio.com/internal/_git/dotnet-release?path=%2Feng%2Fpipeline%2Ftemplates%2Fsteps%2Fsdl-validation.yml&version=GBmaster&line=75&lineEnd=75&lineStartColumn=10&lineEndColumn=32&lineStyle=plain

Comment (sunandabalu) [reply eng/common/sdl/execute-all-sdl-tools.ps1:111]: guardian break will only break the build if the tools find any failures, if guardian itself fails it returns an exit code. If guardian fails we are already failing the leg is my guess but this code you added will not catch it. https://www.1eswiki.com/wiki/Secure_Development_Tools_Guardian#Build_Breaks

Comment (jcagme) [reply eng/common/sdl/execute-all-sdl-tools.ps1:111]: > This is a param plumbed through to be able to break build if necessary. Did you run a test build with this set to true?

Of course! https://dnceng.visualstudio.com/internal/_build/results?buildId=660013&view=logs&j=d5be4b06-a322-5f90-54ec-b848ca73ce8f&t=1122512c-76b8-5e1d-dede-c3a5453dcdb4

Comment (sunandabalu) [reply eng/common/sdl/execute-all-sdl-tools.ps1:111]: Is there a build that failed? Is this the right pipeline - https://dnceng.visualstudio.com/internal/_build/results?buildId=661658&view=results

Comment (jcagme) [reply eng/common/sdl/execute-all-sdl-tools.ps1:111]: What do you mean by "build that failed"?

Comment (sunandabalu) [reply eng/common/sdl/execute-all-sdl-tools.ps1:111]: Github loading is so delayed, it showed me the link you posted for the build failure after I put that comment in 🤦

**Pull request merged**